d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -510,17 +543,35 @@ void sendAckOrNack(ResourceType type, String nonce, List<String> errors)
     }
   }
 
+  private void processResourceChanges(ResourceType type, Map<String, ? extends ResourceUpdate> updates,
+      Collection<String> removedResources)
+  {
+    handleResourceUpdate(updates, type);
+    handleResourceRemoval(removedResources, type);
+    WildcardResourceSubscriber wildcardResourceSubscriber = getWildcardResourceSubscriber(type);
+    if (wildcardResourceSubscriber != null)
+    {
+      // lastly notify wildcard subscriber of the end of the changes
+      wildcardResourceSubscriber.onAllResourcesProcessed();
+    }
+  }
+
   private void handleResourceUpdate(Map<String, ? extends ResourceUpdate> updates, ResourceType type)
   {
     Map<String, ResourceSubscriber> subscribers = getResourceSubscriberMap(type);
+    WildcardResourceSubscriber wildcardSubscriber = getWildcardResourceSubscriber(type);
+
     for (Map.Entry<String, ? extends ResourceUpdate> entry : updates.entrySet())
     {
-      ResourceSubscriber subscriber = subscribers.get(entry.getKey());
-      if (subscriber != null)
+      if (subscribers != null)
```

Comment:
If you update the definition of `_resourceSubscribers` this is now impossible and isn't needed